### PR TITLE
Set serverAuth extended key usage for generated certificates and CSRs.

### LINKED ISSUE #81067: certutil http and auto-configured certificates for http layer should set extended key usage
We generate certificates that are intended to be used for server authentication in a couple of places:

- `bin/elasticsearch-certutil http` command  ( See `CertGenUtils.generateSignedCertificate` in `HttpCertificateCommand` )
- TLS auto-configuraion for node startup ( see [`AutoConfigureNode`](https://github.com/elastic/elasticsearch/blob/814f7f9f52cb8c05db6465b9bfc818ea5d76c575/x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java#L454) )

In these cases we know that the certificate is meant to be used for server authentication, so we can set Extended Key Usage extension to `1.3.6.1.5.5.7.3.1`

### PR DESCRIPTION
**Pull request by slobodanadamovic**
This PR changes `CertGenUtils` to allow configuring
`ExtendedKeyUsage` extension for generated certificates (and CSRs)
and adds `id_kp_serverAuth` (1.3.6.1.5.5.7.3.1) key usage
to places where we know that the certificate is meant 
to be used for server authentication.

Closes #81067

### CODE REVIEW

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertGenUtils.java [hunk 1]
```diff
@@ -270,13 +318,43 @@ private static String getDefaultSignatureAlgorithm(PrivateKey key) {
...
         Objects.requireNonNull(principal, "Principal must not be null");
         JcaPKCS10CertificationRequestBuilder builder = new JcaPKCS10CertificationRequestBuilder(principal, keyPair.getPublic());
+
+        ExtensionsGenerator extGen = new ExtensionsGenerator();
         if (sanList != null) {
-            ExtensionsGenerator extGen = new ExtensionsGenerator();
             extGen.addExtension(Extension.subjectAlternativeName, false, sanList);
+        }
+
+        if (extendedKeyUsages != null) {
```
Comment (albertzaharovits) [inline x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertGenUtils.java:351]: nit: The null check is redundant and ugly for collections.

**Review by albertzaharovits — APPROVED** [review]
LGTM

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertGenUtils.java (hunk 1 above)
Comment (slobodanadamovic) [reply x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertGenUtils.java:351]: Agreed.

**Comment by slobodanadamovic** [comment]
elasticsearch-ci/part-2  seems to be failing on master as well: #86496 86496

**Comment by slobodanadamovic** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Pull request merged**